velox/functions/lib/Re2Functions.cpp

```diff
@@ -496,46 +496,72 @@ bool matchExactPattern(
   return input.size() == 0;
 }
 
+// Matches a kRelaxedFixed pattern forward against ASCII input.
+// Returns a pair:
+//   - first: a bool which tells us whether it matches.
+//   - second: an integer which has different meaning depends on 'first'.
```

Comment:
I couldn't understand this description until I read the code. I wonder if my understanding is correct.

The result of function function might be full match, partial match, no match at all.

Full match (all sub-patterns starting from startSubPatternIndex inclusive and ending at endSubPatternIndex inclusive) is indicated by returning `{true, <index of the first character after the match>}`.

Partial match (one or more sub-patterns at the start of the list match) is indicated by returning `{false, <index of the first sub-pattern that didn't match>}`.

No match is indicated by returning `{false, -1}`.

Is this correct?